### PR TITLE
Zig build

### PR DESCRIPTION
**Pull request by travisstaloch**
this patch allows building with the zig compiler in order to make compiling on other platforms and cross-compilation to other platforms easier.  the zig compiler supports building c/c++ code via libclang on many platforms (https://ziglang.org/download/).  it supports cross compiling to all major platforms by bundling a variety of different libcs.

### CODE REVIEW

**Review by HenryRLee — COMMENTED** [review]
Many thanks for your contribution!

When I was testing the build, I encountered this error:
```
./build.zig:9:19: error: no member named 'addOptions' in struct 'std.build.Builder'
    const opts = b.addOptions();
```

The zig version I used was 0.8.1, downloaded from brew.

Do you know why it'd happen? How can I fix it?

**Comment by travisstaloch** [comment]
This error happens because this api changed post 0.8.1 in august i believe.  You'll need to use `brew install zig --HEAD` for now.  

Unfortunately zig isn't quite stable yet.  So while this particular error won't be a problem after 0.9.0 is released [around jan 14](https://github.com/ziglang/zig/milestones) , there may be other breaking changes before 1.0.  Perhaps a note in the readme explaining this requirement?  Let me know if you'd like any changes or have any other problems.

File: cpp/README.md [hunk 1]
```diff
@@ -20,6 +20,27 @@ Run the `unit_tests` to perform the unit tests:
...
+$ zig build -Drelease-fast
+
+# create an optimized zig-out/lib/libpheval.so (or .dll/.pdb on windows)
+$ zig build -Ddynamic -Drelease-fast
+
+# cross-compile for windows from another platform. creates zig-out/lib/libpheval.dll/pdb
+$ zig build -Dtarget=x86_64-windows -Drelease-fast
+
+# create zig-out/lib/libphevalomaha.a 
+zig build -Domaha -Drelease-fast
```
Comment (HenryRLee) [inline cpp/README.md:41]: Could you please add a '$' symbol in the front, similar to the other command lines?

**Review by HenryRLee — CHANGES_REQUESTED** [review]
Thanks for the explanation. I tested the build in zig 0.9.0 and it works good. Could you please add one more line in the readme noting this minimal version requirement?

File: cpp/README.md (hunk 1 above)
Comment (travisstaloch) [reply cpp/README.md]: Thanks.  I updated the readme.  I also added to the linux ci job so that building with zig is tested.

**Review by HenryRLee — APPROVED** [review]
Everything looks good. Thank you for your contribution!

**Comment by travisstaloch** [comment]
Thanks! I'll keep my fork up until 0.9 lands w/ a weekly ci run to try and catch any breaking zig changes.  Do let me know if you run into any issues or want to add anything.  I was considering adding the unit_tests exe as well, but didn't get around to it.

**Comment by travisstaloch** [comment]
A added weekly ci jobs which build on linux, macos and windows w/ a little bit of cross compilation on each. https://github.com/travisstaloch/PokerHandEvaluator/actions

**Comment by HenryRLee** [comment]
Thanks @travisstaloch, I can see your changes in your branch. Do you want to create a new PR for those changes? Since the previous PR had been merged.

**Pull request merged**